src/common/types/store-data/visualization-store-data.ts

```diff
@@ -22,6 +22,8 @@ export interface VisualizationStoreData {
     selectedDetailsViewPivot: DetailsViewPivotType;
     injectingRequested: boolean;
     injectingStarted: boolean;
+    injectionFailed?: boolean;
```

Comment:
It feels awkward to use 3 booleans for `injectingRequested`, `injectingStarted`, `injectionFailed` when only 4 of the 8 combinations of them represent valid states (`fff`/`tff`/`ttf`/`fft`). Perhaps it would be easier to understand the state machine if these 3 were replaced by a single `injectingState` enum?